ynr/apps/people/forms.py

```diff
@@ -63,6 +60,48 @@ class CandidacyDeleteForm(BaseCandidacyForm):
     )
 
 
+class PersonIdentifierForm(forms.ModelForm):
+    class Meta:
+        model = PersonIdentifier
+        exclude = ("internal_identifier", "extra_data")
+
+    def has_changed(self, *args, **kwargs):
+        """
+        The `has_changed` method tells Django if it should validate and process
+        the form or not.
+
+        In a formset it's used to detect if an "extra" form has content or not,
+        so for example an empty "extra" form don't raise ValidationErrors.
+
+        This method catches the case where someone selects a value_type from
+        a drop down but doesn't add a value. As there is no new content being
```

Comment:
Not for right now, but it seems like a complimentary javascript 'progressive enhancement' to pair with this on the front-end would be to disable `value_type` by default and only enable `value_type` once `value` is populated.

Reply:
Resolving this as you've linked to it in #198